t/ui/14-dashboard.t

```diff
@@ -94,13 +94,23 @@ is(
 
 $driver->find_element_by_link_text('400')->click();
 is($driver->find_element('#more_builds b')->get_text(), 400, 'limited to the selected number');
+like($driver->get_current_url(), qr/1001\?limit_builds=400/, 'url shows the selected build_limits');
+$driver->find_element_by_link_text('50')->click();
+my $actual_url = $driver->get_current_url();
+unlike($actual_url, qr/limit_builds=400/, 'url query doesnt show previous limit_builds');
+like($actual_url, qr/1001\?limit_builds=50/, 'url query has unique limit_builds entry');
 $driver->find_element_by_link_text('tagged')->click();
 is(scalar @{$driver->find_elements('.h4', 'css')}, 0, 'no tagged builds exist');
+$driver->find_element_by_link_text('400')->click();
+like(
+    $driver->get_current_url(),
+    qr/1001\?only_tagged=1&(?!.*limit_builds&)limit_builds=400/,
```

Comment:
I'm wondering about the middle part here (`(?!.*limit_builds&)`). This is again a bit too much regex magic for my taste and I would use `unlike` as before.

It would also make sense test for `only_tagged=1` separately to make this check not assert the order in which these parameters are present.